src/generate/formats/readers/openapi/inputs.ts

```diff
@@ -120,13 +125,28 @@ const buildBodyInputs = (
     .filter(([, prop]) => !prop.readOnly) // Don't create inputs for readonly properties
     .map<Input>(([propKey, prop]) => {
       const schemaType = prop?.type;
-      const { type, cleanFn, cleanReturnType, allowOptional } =
-        toInputType[schemaType as string] ?? toInputType.string;
+      let inputType: InputType;
+
+      if (schemaType === "array") {
+        inputType = formatArrayInput(
+          prop as OpenAPIV3.ArraySchemaObject | OpenAPIV3_1.ArraySchemaObject,
+        );
+      } else {
+        inputType = toInputType[schemaType as string] ?? toInputType.string;
+      }
+
+      const { type, cleanFn, cleanReturnType, allowOptional, collection } = inputType;
 
       const required = requiredKeys.has(propKey) || !allowOptional;
-      const clean = required
-        ? `(value): ${cleanReturnType} => util.types.${cleanFn}(value)`
-        : `(value): ${cleanReturnType} | undefined => value !== undefined && value !== null ? util.types.${cleanFn}(value) : undefined`;
+      let clean: string | undefined;
+
+      if (schemaType === "array") {
+        clean = `(values) => ((values as string[]) || []).map((value) => util.types.${cleanFn}(value))`;
```

Comment:
I think `as unknown[]` would be more accurate as the first arg is `unknown`. I could also see an argument for adding `Array.isArray` if we wanted to avoid such casts.